infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java

```diff
@@ -33,13 +61,210 @@
  */
 @Listeners(MockitoTestNGListener.class)
 public class OpenShiftProjectFactoryTest {
+
+  @Mock private OpenShiftClientConfigFactory configFactory;
   @Mock private OpenShiftClientFactory clientFactory;
+
+  @Mock
+  private NonNamespaceOperation<
+          Project, ProjectList, DoneableProject, Resource<Project, DoneableProject>>
+      projectOperation;
+
+  @Mock private OpenShiftClient osClient;
+
   private OpenShiftProjectFactory projectFactory;
 
+  @BeforeMethod
+  public void setUp() throws Exception {
+    lenient().when(clientFactory.createOC()).thenReturn(osClient);
+    lenient().when(osClient.projects()).thenReturn(projectOperation);
+  }
+
+  @Test(
+      expectedExceptions = ConfigurationException.class,
+      expectedExceptionsMessageRegExp =
+          "che.infra.kubernetes.namespace.default or "
+              + "che.infra.kubernetes.namespace.allow_user_defined must be configured")
+  public void
+      shouldThrowExceptionIfNoDefaultNamespaceIsConfiguredAndUserDefinedNamespacesAreNotAllowed()
+          throws Exception {
+    projectFactory =
+        new OpenShiftProjectFactory(
+            "projectName", "", "", null, false, clientFactory, configFactory);
+  }
+
+  @Test
+  public void shouldReturnDefaultProjectWhenItExistsAndUserDefinedIsNotAllowed() throws Exception {
+    prepareNamespaceToBeFoundByName(
+        "che-default",
+        new ProjectBuilder()
+            .withNewMetadata()
+            .withName("che-default")
+            .withAnnotations(
+                ImmutableMap.of(
+                    PROJECT_DISPLAY_NAME_ANNOTATION,
+                    "Default Che Project",
+                    PROJECT_DESCRIPTION_ANNOTATION,
+                    "some description"))
```

Comment:
For completeness, consider also testing case where display name and description are null.

Reply:
It's already handled in https://github.com/eclipse/che/blob/3b3eeb5f66ce00c9523fcb8486c64e1de3232c84/infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactoryTest.java#L171

Do you think it makes sense to move it to a separate test method?

Reply:
No, as long as it's covered I think it's fine -- I missed that part in the other method.